api/src/main/scala/hmda/api/http/public/HmdaFilerPaths.scala

```diff
@@ -57,6 +64,50 @@ trait HmdaFilerPaths extends HmdaCustomDirectives with HmdaFilerProtocol with Ap
       }
   }
 
+  def hmdaFilerMsasPath(supervisor: ActorRef) = {
+    path("filers" / Segment / Segment / "msaMds") { (period, instId) =>
+      encodeResponse {
+        timedGet { uri =>
+          val fSubmissions = (supervisor ? FindSubmissions(SubmissionPersistence.name, instId, period)).mapTo[ActorRef]
+          val hmdaFilerPersistenceF = (supervisor ? FindHmdaFilerPersistence(HmdaFilerPersistence.name)).mapTo[ActorRef]
+
+          val fMsas = for {
+            subActor <- fSubmissions
+            submissions <- (subActor ? GetState).mapTo[Seq[Submission]]
+
+            manager <- (supervisor ? FindProcessingActor(SubmissionManager.name, getLatestSignedSub(submissions))).mapTo[ActorRef]
+            larStats <- (manager ? GetActorRef(SubmissionLarStats.name)).mapTo[ActorRef]
+            stats <- (larStats ? FindIrsStats(getLatestSignedSub(submissions))).mapTo[Seq[Msa]]
+
+            a <- hmdaFilerPersistenceF
+            filer <- (a ? FindHmdaFiler(instId)).mapTo[Option[HmdaFiler]]
+          } yield (stats, filer)
+
+          onComplete(fMsas) {
+            case Success((msas, instOpt)) =>
+              val msasJson = msas.filterNot(_.id == "NA").map(msa => s"""{"id":"${msa.id}","name":"${msa.name}"}""").mkString("[", ",", "]")
+              val inst = instOpt.getOrElse(HmdaFiler("", "", "", ""))
+              if (inst.institutionId == "") {
+                complete(ToResponseMarshallable(StatusCodes.NotFound -> s"Unable to find institution $instId"))
+              } else {
+                val response =
+                  s"""
+                       |{
+                       |  "year": "$period",
+                       |  "institution": {"name":"${inst.name}", "id":"${inst.institutionId}", "respondentId": "${inst.respondentId}"},
+                       |  "msaMds": $msasJson
+                       |}
+                   """.stripMargin.parseJson
```

Comment:
I'm going to merge this as is, but I don't think this is a good style for producing JSON output. Moving forward we should keep the same conventions that we have used in the project, by serializing/deserializing to case classes.